src/components/dms/NewChat.tsx

```diff
@@ -0,0 +1,248 @@
+import {AppBskyActorDefs, moderateProfile} from '@atproto/api'
+import {BottomSheetFlatListMethods} from '@discord/bottom-sheet'
+import {msg, Trans} from '@lingui/macro'
+import {useLingui} from '@lingui/react'
+
+import {sanitizeDisplayName} from '#/lib/strings/display-names'
+import {sanitizeHandle} from '#/lib/strings/handles'
+import {isWeb} from '#/platform/detection'
+import {useActorSearch} from '#/state/queries/actor-search'
+import {useModerationOpts} from '#/state/queries/preferences'
+import {FAB} from '#/view/com/util/fab/FAB'
+import * as Toast from '#/view/com/util/Toast'
+import {UserAvatar} from '#/view/com/util/UserAvatar'
+import {atoms as a, useTheme, web} from '#/alf'
+import * as Dialog from '#/components/Dialog'
+import * as TextField from '#/components/forms/TextField'
+import {useThrottledValue} from '#/components/hooks/useThrottledValue'
+import {MagnifyingGlass2_Stroke2_Corner0_Rounded as Search} from '#/components/icons/MagnifyingGlass2'
+import {useGetChatFromMembers} from '../../screens/Messages/Temp/query/query'
+import {Button} from '../Button'
+import {Envelope_Stroke2_Corner0_Rounded as Envelope} from '../icons/Envelope'
+import {ListMaybePlaceholder} from '../Lists'
+import {Text} from '../Typography'
+
+export function NewChat({onNewChat}: {onNewChat: (chatId: string) => void}) {
+  const control = Dialog.useDialogControl()
+  const t = useTheme()
+  const {_} = useLingui()
+
+  const {mutate: createChat} = useGetChatFromMembers({
+    onSuccess: data => {
+      onNewChat(data.chat.id)
+    },
+    onError: error => {
+      Toast.show(error.message)
+    },
+  })
+
+  const onCreateChat = useCallback(
+    (did: string) => {
+      control.close(() => createChat([did]))
+    },
+    [control, createChat],
+  )
+
+  return (
+    <>
+      <FAB
+        testID="newChatFAB"
+        onPress={control.open}
+        icon={<Envelope size="xl" fill={t.palette.white} />}
+        accessibilityRole="button"
+        accessibilityLabel={_(msg`New chat`)}
+        accessibilityHint=""
+      />
+
+      <Dialog.Outer
+        control={control}
+        testID="newChatDialog"
+        nativeOptions={{sheet: {snapPoints: ['100%']}}}>
+        <Dialog.Handle />
+        <SearchablePeopleList onCreateChat={onCreateChat} />
+      </Dialog.Outer>
+    </>
+  )
+}
+
+function SearchablePeopleList({
+  onCreateChat,
+}: {
+  onCreateChat: (did: string) => void
+}) {
+  const t = useTheme()
+  const {_} = useLingui()
+  const moderationOpts = useModerationOpts()
+  const control = Dialog.useDialogContext()
+  const listRef = useRef<BottomSheetFlatListMethods>(null)
+
+  const [search, setSearch] = useState('')
+  const throttledSearch = useThrottledValue(search, 300)
+
+  const isSearching = throttledSearch.length > 0
+
+  const searchQuery = useActorSearch({
+    query: throttledSearch || 'temp',
+  })
+
+  const {data, isLoading, isError, refetch} = isSearching
+    ? searchQuery
+    : searchQuery
```

Comment:
Oo, we recently used `keepPreviousData` for this in the search screen, I think it would be great here too. I just modified the query hook to take in a bool for if we want to keep the prev data, see: https://github.com/bluesky-social/social-app/blob/main/src/state/queries/actor-autocomplete.ts#L19

Then we implemented like this: https://github.com/bluesky-social/social-app/blob/main/src/view/screens/Search/Search.tsx#L811

We can probably just get rid of the throttle too here I think. So combined, we'll only get a spinner on the first character of the query instead of disappearing results.

Reply:
Oh, actually I think we can just use the existing `useActorAutocompleteQuery()` here?

Reply:
Huh, I did some testing and the `searchActors` query is really unreliable anyway. Sometimes I don't get any results for something that _definitely_ should return something, like `haileyok`. I do get the results though in the typeahead.

I think after we merge this https://github.com/bluesky-social/social-app/pull/3765 that the typeahead will be the most reliable here.